### PR TITLE
C++: Drop redundant columns from `files` and `folders` relations

### PR DESCRIPTION
**Pull request by hvitved**
Replaces https://github.com/github/codeql/pull/6477.

### CODE REVIEW

**Review by geoffw0 — COMMENTED** [review]
LGTM.

One point to discuss.

Have you tested the upgrade script / is there CI for this these days?

File: cpp/ql/lib/semmle/code/cpp/File.qll [hunk 1]
```diff
@@ -336,7 +326,13 @@ class File extends Container, @file {
...
    * "tar.gz", while `getExtension` will have the result "gz".
    */
-  string getExtensions() { files(underlyingElement(this), _, _, result, _) }
+  string getExtensions() {
+    exists(string name, int firstDotPos |
+      name = this.getBaseName() and
+      firstDotPos = min([name.indexOf("."), name.length() - 1]) and
+      result = name.suffix(firstDotPos + 1)
+    )
+  }
```
[the remark this answers is not in the document]
Comment (hvitved) [reply cpp/ql/lib/semmle/code/cpp/File.qll:335]: I thought it was better to make the implementation match the spec, which is why I didn't change it. If we change it, there will be no difference between `getExtension` and `getExtensions`, will there?

**Comment by geoffw0** [comment]
The language test failures are expected as they are run without the corresponding semmle-code changes (where the tests pass).  It looks like the integration test failures are of a similar nature, but please speak up if you disagree.

**Review by geoffw0 — APPROVED** [review]
:+1:

**Pull request merged**